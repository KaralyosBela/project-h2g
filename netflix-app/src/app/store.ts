import { configureStore } from "@reduxjs/toolkit";
import moviesReducer from "../features/moviesSlice";
import hehe from "../features/actions"

export const store = configureStore({
  reducer: {
    movies: moviesReducer,
    bruh: hehe
  }
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
