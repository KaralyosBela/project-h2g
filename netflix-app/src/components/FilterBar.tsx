import classes from "./FilterBar.module.css";

export const FilterBar: React.FC = () => {
  return (
    <div>
      <div className={classes.filterBar}>
        <ul className={classes.categories}>
          <li>All</li>
          <li>Documentary</li>
          <li>Comedy</li>
          <li>Horror</li>
          <li>Crime</li>
        </ul>
        <div className={classes.sortSection}>
          <div>Sort by</div>
          <select id="sortby">
            <option value="1">Release date</option>
            <option value="2">Release date</option>
          </select>
        </div>
      </div>
      <hr/>
      <div className={classes.movieCounter}>
        <span>50</span> movies found
      </div>
    </div>
  );
};
