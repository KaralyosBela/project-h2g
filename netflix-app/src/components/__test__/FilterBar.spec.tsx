import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { FilterBar } from "../FilterBar";
import { Provider } from "react-redux";
import { store } from "../../app/store";
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

const MockFilterBar = ({movieCount}: any) => {
  return (
    <Provider store={store}>
      <FilterBar movieCount={movieCount} />
    </Provider>
  );
};

it("should render the filterbar component correctly", () => {
  const {container} = render(<MockFilterBar movieCount={10} />);
  expect(container).toMatchSnapshot();
  // screen.debug();
});

it("should render the filterbar component with the correct amount of movies", () => {
  render(<MockFilterBar movieCount={900} />);
  const spanElement = screen.getByText(/900/i);
  expect(spanElement).toBeInTheDocument();
  // screen.debug();
});

it("the 'all' button should be selected on init", () => {
  render(<MockFilterBar movieCount={0} />);
  const allButtion = screen.getByText(/All/i);
  expect(allButtion).toHaveClass("active");
  // screen.debug();
});

it("the 'comedy' button should have an active class when clicked", () => {
  render(<MockFilterBar movieCount={0} />);
  const allButton = screen.getByText(/All/i);
  const actionButton = screen.getByText(/Action/i);
  const comedyButton = screen.getByText(/Comedy/i);
  const horrorButton = screen.getByText(/Horror/i);
  const crimeButton = screen.getByText(/Crime/i);

  fireEvent.click(allButton);
  expect(allButton).toHaveClass("active");
  fireEvent.click(actionButton);
  expect(actionButton).toHaveClass("active");
  fireEvent.click(comedyButton);
  expect(comedyButton).toHaveClass("active");
  fireEvent.click(horrorButton);
  expect(horrorButton).toHaveClass("active");
  fireEvent.click(crimeButton);
  expect(crimeButton).toHaveClass("active");
  // screen.debug();
});

