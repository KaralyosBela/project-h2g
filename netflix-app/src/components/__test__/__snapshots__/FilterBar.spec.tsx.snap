// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the filterbar component correctly 1`] = `
<div>
  <div>
    <div
      class="filterBar"
    >
      <div
        class="categories"
      >
        <button
          class="active"
        >
          All
        </button>
        <button
          class=""
        >
          Action
        </button>
        <button
          class=""
        >
          Comedy
        </button>
        <button
          class=""
        >
          Horror
        </button>
        <button
          class=""
        >
          Crime
        </button>
      </div>
      <div
        class="sortSection"
      >
        <div>
          Sort by
        </div>
        <select>
          <option
            value="releaseDate_ascending"
          >
            Release date asc
          </option>
          <option
            value="releaseDate_descending"
          >
            Release date desc
          </option>
          <option
            value="length_ascending"
          >
            Length asc
          </option>
          <option
            value="length_descending"
          >
            Length desc
          </option>
          <option
            value="rating_ascending"
          >
            Rating asc
          </option>
          <option
            value="rating_descending"
          >
            Rating desc
          </option>
        </select>
      </div>
    </div>
    <div
      class="movieCounter"
    >
      <span>
        10
      </span>
       movies found
    </div>
  </div>
</div>
`;
